libraries/shared/storage/diff_repository.go

```diff
@@ -73,3 +74,11 @@ func (repository diffRepository) MarkChecked(id int64) error {
 	_, err := repository.db.Exec(`UPDATE public.storage_diff SET checked = true WHERE id = $1`, id)
 	return err
 }
+
+func (repository diffRepository) GetFirstDiffForBlockHeight(blockHeight int64) (types.PersistedDiff, error) {
+	var diff types.PersistedDiff
+	err := repository.db.Get(&diff,
+		`SELECT * FROM public.storage_diff WHERE checked IS false AND block_height = $1 LIMIT 1`, blockHeight)
```

Comment:
wondering if it might make sense to say `block_height >= $1 ORDER BY block_height ASC LIMIT 1` in case we don't have a diff at exactly that block height

Reply:
ah, good catch!